Packet++/src/Asn1Codec.cpp

```diff
@@ -621,7 +621,6 @@ namespace pcpp
 	std::vector<uint8_t> Asn1IntegerRecord::encodeValue() const
 	{
 		std::vector<uint8_t> result;
-		result.reserve(m_ValueLength);
```

Comment:
I have no idea why the compiler complains here, but I think it's fine to remove it:
https://github.com/seladb/PcapPlusPlus/actions/runs/9562329238/job/26358481698

Reply:
can we ignore this line only for `MinGW`?

Reply:
It's a compilation error so not sure we can ignore it, but I think removing it is ok, I don't foresee a big performance impact because of it

Reply:
I mean `ifndef MinGW` or some similar syntax. 
I think it's better to keep the `reserve` as it still works for other platforms.